### PR TITLE
Csp 1094 put calendar event

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        };
+
+        UpdateCalendarEventToNewValues(command, existingEvent);
+
+        audit.After = JsonSerializer.Serialize(existingEvent);
+        _auditWriteRepository.Create(audit);
+
+        if (command.SendUpdateEventNotification)
+        {
+            var existingAttendances =
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandler.cs:56]: We are not updating attendance so this should use read repository

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/ValidateCalendarEventId.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            result.ShouldNotHaveValidationErrorFor(c => c.CalendarEventId);
+            result.ShouldHaveAnyValidationError();
+        }
+        else
+        {
+            result
+                .ShouldHaveValidationErrorFor(c => c.CalendarEventId)
+                .WithErrorMessage(errorMessage);
+            // result.ShouldNotHaveValidationErrorFor(c => c.CalendarId);
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/ValidateCalendarEventId.cs:35]: 💥

File: src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandler.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandler.cs:56]: nice spot!

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    public async Task Handle_PutCalendarEvent_SendNotificationFalse(
+    [Frozen] Mock<IAanDataContext> aanDataContext,
+    [Frozen] Mock<ICalendarEventsWriteRepository> calendarEventsWriteRepository,
+    [Frozen] Mock<IMembersReadRepository> membersReadRepository,
+    [Frozen] Mock<INotificationsWriteRepository> notificationsWriteRepository,
+    [Frozen] Mock<IAuditWriteRepository> auditWriteRepository,
+    [Frozen] PutCalendarEventModel model,
+    [Frozen] List<Member> members,
+    Guid calendarEventId,
+    Guid memberId)
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandlerTests.cs:78]: Indention please

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/PutCalendarEventCommandValidatorBuilder.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/PutCalendarEventCommandValidatorBuilder.cs:1]: A lot of commented code here

File: src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandler.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Application/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandler.cs:56]: I have addressed this, but for some reason GitHub is not marking it as outdated, so leaving it 'unresolved' so you can check.

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandlerTests.cs (hunk 3 above)
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandHandlerTests.cs:78]: ok

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/PutCalendarEventCommandValidatorBuilder.cs
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/PutCalendarEventCommandValidatorBuilder.cs:1]: removed

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/ValidateCalendarEventId.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+using ErrorConstants = SFA.DAS.AANHub.Application.CalendarEvents.Commands.PutCalendarEvent.PutCalendarEventCommandValidator;
+
+namespace SFA.DAS.AANHub.Application.UnitTests.CalendarEvents.Commands.PutCalendarEvent.PutCalendarEventCommandValidatorTests;
+
+public class ValidateCalendarEventId
+{
+    [TestCase("00000000-0000-0000-0000-000000000000", false, ErrorConstants.EventIdNotSuppliedMessage)]
+    [TestCase("c7be8d81-b811-43f2-bd92-e48a2724d7d3", false, ErrorConstants.EventNotFoundMessage)]
+    [TestCase("d19b813b-b5f2-40f1-b9f7-fdaf488283b7", false, ErrorConstants.EventNotActiveMessage)]
+    [TestCase("b09c88f7-01b6-48fe-a755-e9591d0da729", false, ErrorConstants.EventInPastMessage)]
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/ValidateCalendarEventId.cs:13]: You have declared these as constants in the helper, so I guess you can use them here?

**Review by DevenKShah — DISMISSED** [review]

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/ValidateCalendarEventId.cs (hunk 4 above)
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Commands/PutCalendarEvent/PutCalendarEventCommandValidatorTests/ValidateCalendarEventId.cs]: ok sounds good

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**